modules/programs/wezterm.nix

```diff
@@ -0,0 +1,545 @@
+            The mouse binding event.
+          '';
+          example = "Left";
+        };
+
+        count = mkOption {
+          type = int;
+          default = 1;
+          description = ''
+            The event count to trigger action.
+          '';
+          example = 1;
+        };
+
+        action = mkBindingAction (literalExample ''"OpenLinkAtMouseCursor"'');
+      };
+    };
+
+  mkColor = default: desc:
+    mkOption {
+      type = types.str;
+      default = default;
+      description = desc;
+      example = default;
+    };
+
+  weztermColorschemeCursorType = with types;
+    submodule {
+      options = {
+        foreground = mkColor "#000000" ''
+          Overrides the text color when the current cell is occupied by the cursor.
+        '';
+        background = mkColor "#52AD70" ''
+          Overrides the cell background color when the current cell is occupied
+          by the cursor and the cursor style is set to <literal>Block</literal>.
+        '';
+        border = mkColor "#52AD70" ''
+          Specifies the border color of the cursor when the cursor style is
+          set to <literal>Block</literal>.
+        '';
+      };
+    };
+
+  weztermColorschemeSelectionType = with types;
+    submodule {
+      options = {
+        foreground = mkColor "#000000" "The foreground color of selected text.";
+        background = mkColor "#FFFACD" "The background color of selected text.";
+      };
+    };
+
+  weztermColorschemeTabBarTabType = with types;
+    tabType:
+    submodule {
+      options = {
+        background = let
+          default = if tabType == "active" then
+            "#2B2042"
+          else if tabType == "inactive" then
+            "#1B1032"
+          else
+            "#3B3052";
+        in mkColor default "The color of the background area for the tab.";
+
+        foreground = let
+          default = if tabType == "active" then
+            "#C0C0C0"
+          else if tabType == "inactive" then
+            "#808080"
+          else
+            "#909090";
+        in mkColor default "The color of the text for the tab.";
+
+        intensity = mkOption {
+          type = enum [ "Half" "Normal" "Bold" ];
+          default = "Normal";
+          description = ''
+            Specify whether you want <literal>Half</literal>, <literal>Normal</literal>
+            or <literal>Bold</literal> intensity for the label shown for this tab.
+          '';
+          example = literalExample "Normal";
+        };
+
+        underline = mkOption {
+          type = enum [ "None" "Single" "Double" ];
+          default = "None";
+          description = ''
+            Specify whether you want <literal>None</literal>, <literal>Single</literal>
+            or <literal>Double</literal> underline for label shown for this tab.
+          '';
+          example = literalExample "None";
+        };
+
+        italic = mkOption {
+          type = bool;
+          default = tabType == "inactive_hover";
```

Comment:
This is a problem since it will make the documentation depend on the user's configuration. Potentially causing documentation rebuilds when the user changes their configuration.

If a default depends on the configuration then there must be a `defaultText` attribute that provides a textual default that does not have such a dependency.